loader_hub/dadjokes/base.py

```diff
@@ -0,0 +1,32 @@
+"""icanhazdadjoke based dad-joke generator."""
+
+import requests
+from typing import List
+
+from gpt_index.readers.base import BaseReader
+from gpt_index.readers.schema.base import Document
+
+class DadJokeReader(BaseReader):
+    """Dad joke reader.
+
+    Reads a random dad joke.
+
+    """
+
+    def _get_random_dad_joke(self):
+        response = requests.get("https://icanhazdadjoke.com/", headers={"Accept": "application/json"})
+        if response.status_code == 200:
+            json_data = response.json()
+            return json_data["joke"]
+        else:
+            return "Sorry, I couldn't get a joke right now. Please try again later."
```

Comment:
Let's throw an error here instead of returning something that actually gets turned into a Document